src/EditorFeatures/CSharp/QuickInfo/SyntacticQuickInfoProvider.cs

```diff
@@ -59,12 +60,20 @@ internal class SyntacticQuickInfoProvider : AbstractQuickInfoProvider
                 return null;
             }
 
-            // If the open brace is the first token of the node (like in the case of a block node or
-            // an accessor list node), then walk up one higher so we can show more useful context
-            // (like the method a block belongs to).
-            if (parent.GetFirstToken() == openBrace)
+            var spanStart = parent.SpanStart;
+            var spanEnd = openBrace.Span.End;
+
+            // If the parent is a scope block, check and include nearby comments around the open brace
```

Comment:
Assuming that "floating" means "top-level", yes I believe they're allowed in ```SourceCodeKind.Script```.  @tmat can confirm (or you could try it :)).